hubblestack/files/hubblestack_nova/mount.py

```diff
@@ -0,0 +1,228 @@
+                ret['mount'][topkey] = []
+            for key, val in data['mount'][topkey].iteritems():
+                if profile and isinstance(val, dict):
+                    val['nova_profile'] = profile
+                ret['mount'][topkey].append({key: val})
+    return ret
+
+
+def _get_tags(data):
+    '''
+    Retrieve all the tags for this distro from the yaml
+    '''
+
+    ret = {}
+    distro = __grains__.get('osfinger')
+
+    for toplist, toplevel in data.get('mount', {}).iteritems():
+        # mount:blacklist
+        for audit_dict in toplevel:
+            # mount:blacklist:0
+            for audit_id, audit_data in audit_dict.iteritems():
+                # mount:blacklist:0:telnet
+                tags_dict = audit_data.get('data', {})
+                # mount:blacklist:0:telnet:data
+                tags = None
+                for osfinger in tags_dict:
+                    if osfinger == '*':
+                        continue
+                    osfinger_list = [finger.strip() for finger in osfinger.split(',')]
+                    for osfinger_glob in osfinger_list:
+                        if fnmatch.fnmatch(distro, osfinger_glob):
+                            tags = tags_dict.get(osfinger)
+                            break
+                    if tags is not None:
+                        break
+                # If we didn't find a match, check for a '*'
+                if tags is None:
+                    tags = tags_dict.get('*', [])
+                # mount:blacklist:0:telnet:data:Debian-8
+                if isinstance(tags, dict):
+                    # malformed yaml, convert to list of dicts
+                    tmp = []
+                    for name, tag in tags.iteritems():
+                        tmp.append({name: tag})
+                    tags = tmp
+                for item in tags:
+                    for name, tag in item.iteritems():
+                        tag_data = {}
+                        # Whitelist could have a dictionary, not a string
+                        if isinstance(tag, dict):
+                            tag_data = copy.deepcopy(tag)
+                            tag = tag_data.pop('tag')
+                        if tag not in ret:
+                            ret[tag] = []
+                        formatted_data = {'name': name,
+                                          'tag': tag,
+                                          'module': 'mount',
+                                          'type': toplist}
+                        formatted_data.update(tag_data)
+                        formatted_data.update(audit_data)
+                        formatted_data.pop('data')
+                        ret[tag].append(formatted_data)
+    return ret
+
+def _check_mount_attribute(path,attribute, check_type):
+    '''
+    This function check if the partition at a given path is mounted with a particular attribute or not. Path to the partition is given in 'path' arguement.
```

Comment:
Lets try to break into multiple lines rather than long lines.
If possible we can reduce the doc string as well, as this is too long.